✨ feat : Spring AI 기반 백엔드 환경 초기 구축 및 AI 적성 테스트 구조 추가

sooyeoneo opened the pull request:
### ☀️ 작업 목적
- AI 적성 추천 및 지역 특화 등 서비스 전체 AI 기능 구현을 위한 Spring AI 기반 백엔드 환경을 구축했습니다.

- OpenAI(GPT-4o) 연동, 공통 설정, 프롬프트 관리, 예외 처리 등 향후 확장 가능한 구조로 설계하였습니다.

---

### 📄 주요 작업 내역
- **Spring AI 1.0.0 의존성 및 spring-milestone BOM 추가**

- `application.yml`에 OpenAI API Key, 모델명, 타임아웃, 재시도, 로깅 등 환경변수/정책 추가

- Spring AI 관련 프로퍼티 자동완성 지원을 위한 `configuration-processor` 추가

- AiConfig: `@EnableRetry` 포함, ChatClient `@Bean` 등록 등 AI Client 기본 설정

- PromptManager: 프롬프트 파일 로드 및 타입별 캐싱/관리 기능 구현

- PromptType enum 정의 (적성검사, 지역특화, 프로그램요약 등)

- BaseAiService: AI 공통 호출/재시도/에러코드 관리 추상 클래스 설계

- AiTestController, AiTestService: AI 연동/테스트용 엔드포인트 및 헬스체크 API 구현
  - `/ai/test/chat` 기본 챗 API, `/ai/test/health` 상태 확인 API

- ErrorCode: AI 관련 에러코드 추가 및 공통 예외 처리 구조 반영

- ai-test.http: 기능 검증을 위한 HTTP 요청 샘플 파일 추가

- aptitude-test.txt: AI 적성 프롬프트 예시 작성 및 프롬프트 관리 구조 도입

---

### 🛠️ 추가 수정 작업 
- 프롬프트 파일 로딩 방식 개선
  - `resource.getFile()`에서 `resource.getInputStream()` 방식으로 변경하여, 
  JAR 패키징/운영환경에서도 정상 동작하도록 수정
  
  - 파일 인코딩을 `UTF-8`로 명시해 OS별 인코딩 이슈 예방

---

### 📌 참고 및 TODO
- 요청/응답 타임아웃 및 오류 fallback 정책(30초, 재시도 등) 적용 중, 추가 테스트 필요

- AI API 요청/응답 로그 및 모니터링 시스템 연동 작업 추후 진행 예정

- 프롬프트 및 AI 응답 형식, 예외 처리 방식 지속 개선

---

#### ⚙️ 기타
- API Key 등 민감정보는 환경변수 및 `.gitignore` 적용 완료

- Spring AI 공식문서 및 실무 표준에 따라 구조 설계

---

**리뷰 및 피드백 환영합니다! 🤗**

sooyeoneo committed:
🏗️ chore : Spring AI 초기 설정

- Spring AI OpenAI 스타터 의존성 추가 (1.0.0-M3)
- application.yml에서 자동완성이나 compile 시 프로퍼티 바인딩 체크를 위한 configuration-processor 추가

sooyeoneo committed:
🔧 add : application.properties에 OpenAI API 설정 추가

- API 키 환경변수 처리
- LLM 모델 GPT-4o로 설정
- 재시도 정책 설정
- 타임아웃 설정 (단위: s → 초 단위로 입력)
- 로깅 레벨 설정 (DEBUG for Spring AI)
- 애플리케이션 프롬프트 경로 설정

sooyeoneo committed:
🏗️ chore : Spring AI 의존성 spring-milestone 추가

sooyeoneo committed:
💡 comment : ? 제거

sooyeoneo committed:
✨ feat : AI 설정 클래스 및 Prompt Manager 구현

   - AiConfig: @EnableRetry 추가 및 기본 옵션 구성
   - PromptManager: 프롬프트 파일 로드 및 캐싱 기능
   - PromptType enum: 적성검사, 지역특화, 프로그램요약 타입 정의

sooyeoneo committed:
🏗️ chore : Spring AI 의존성 추가

- spring-ai-openai-spring-boot-starter 버전 수정 및 BOM 설정 추가
- snapshot 추가
- Spring Retry 의존성 추가

sooyeoneo committed:
🔧 add : AiConfig에 ChatClient 빈 설정 추가

sooyeoneo committed:
✨ feat : AI 서비스 기본 구조 및 예외 처리 구현

- BaseAiService: AI 호출 공통 로직 및 재시도 정책
- ErrorCode: AI 관련 error code 추가

sooyeoneo committed:
✅  test : AiTestController, AiTestService 추가

- AiTestController 구현
  - /ai/test/chat: 기본 AI 채팅 테스트 엔드포인트
  - /ai/test/health: 서비스 상태 헬스 체크 API 추가
- AiTestService 구현 (BaseAiService 상속)
- BaseAiService 추상 클래스 작성 (AI 공통 로직 담당)

sooyeoneo committed:
📝  docs : AI 테스트용 http 파일 및 적성 테스트 프롬프트 추가

- ai-test.http: AI 동작 및 챗 테스트용 HTTP 요청 샘플 파일 추가
- aptitude-test.txt: AI 적성 상담 프롬프트 예시 문서 작성

jiyoon0000 commented on src/main/java/com/salayo/locallifebackend/domain/ai/prompt/PromptManager.java:
```diff
@@ -0,0 +1,37 @@
+	private String loadPrompt(PromptType type) {
+		try {
+			Resource resource = resourceLoader.getResource(basePath + type.getFileName());
+			return new String(Files.readAllBytes(resource.getFile().toPath()));
```
제가 알기로 나중에 JAR로 패키징해서 배포하면 리소스 파일이 파일 시스템이 아니라 클래스패스 리소스로 바뀌기 때문에 `getFile()`을 사용하게 되면 FileNotFoundException이 발생할 수 있을 것 같아요!🥲

그래서 찾아보니까 `getFile()` 방식이 아니라 `resource.getInputStream()`을 사용하는 방식으로 바꾸면 배포환경에서도 잘 동작한다고 하네요!☺️

sooyeoneo replied:
오 감사합니다 당장 수정할게요!!! 😎

sooyeoneo committed:
🐛 fix: 프롬프트 파일 로딩 시 InputStream 및 UTF-8 인코딩 적용

- resource.getFile() → resource.getInputStream() 방식으로 변경
- JAR 배포 환경에서 파일 읽기 오류 방지
- 파일 인코딩을 UTF-8로 명시하여 OS별 인코딩 이슈 해결

jiyoon0000 commented:
구웃🩷
